build.gradle

```diff
@@ -124,39 +139,197 @@ publishing {
+        // Defaults to `NEVER`.
+        active = 'ALWAYS'
+
+        // Generates an armored signature.
+        // Defaults to `false`.
+        //
+        // We set this to `true` to produce human-readable `.asc` signature files.
+        // NOTE: Signing fails with "Unexpected error" when set to `false`.
+        armored = true
+
+        // Verify signature files.
+        // If `false` then `publicKey` may be omitted.
+        // Defaults to `true`.
+        //
+        // NOTE: Disabled because there's no need to verify signatures immediately after signing.
+        // If we want to enable this in the future, we'll need to configure `publicKey` to our PGP
+        // public key.
+        verify = false
+
+        // The passphrase required to read secret keys.
+        // Set via env var `JRELEASER_GPG_PASSPHRASE`.
+        //
+        // NOTE: We set this using a project property. See
+        // https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties
+        // NOTE: This raises an error if left blank, however this is a bug that should be fixed in
+        // the 1.19.0 release. See https://github.com/jreleaser/jreleaser/issues/1883.
+        passphrase = findProperty("signingPassword") ?: '__DUMMY_VALUE__'
+
+        // The private GPG (ascii armored) used to sign files and commits.
+        // Required when mode = `MEMORY` || `FILE`.
+        // Set via env var `JRELEASER_GPG_SECRET_KEY`.
+        //
+        // NOTE: We set this using a project property. See
+        // https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties
+        // NOTE: If set to a non-empty malformed value, signing will fail with "Unexpected error".
+        secretKey = findProperty("signingKey")
+    }
+
+    // Artifact deployment configuration.
+    deploy {
+        // Deploy staged artifacts to target Maven compatible repositories.
+        maven {
+            // Deploy staged artifacts to Maven Central using the publisher API.
+            // See https://jreleaser.org/guide/latest/reference/deploy/maven/maven-central.html
+            mavenCentral {
+                // The name of the deployer (deployers require a name).
+                app {
+                    // Enables or disables the deployer.
+                    // Supported values are [`NEVER`, `ALWAYS`, `RELEASE`, `SNAPSHOT`].
+                    // Defaults to `NEVER`.
+                    active = 'ALWAYS'
+
+                    // URL where the MavenCentral service is enabled.
+                    url = 'https://central.sonatype.com/api/v1/publisher'
+
+                    // The username required for authorization.
+                    // Set via env var `JRELEASER_MAVENCENTRAL_USERNAME`.
+                    //
+                    // NOTE: We set this using a project property. See
+                    // https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties
```

Comment:
I don't understand why we are linking this article. It explains what gradle properties are, which are a relatively well known concept. Is there something specific in there which explain something as to why?
The main thing to capture here is the WHY. Also, does setting the env var `JRELEASER_MAVENCENTRAL_USERNAME` override this setting when we use properties like this?